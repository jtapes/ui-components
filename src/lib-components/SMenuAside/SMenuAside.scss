@import '../../theme/styles/variables';
@import '../../theme/styles/mixins';
@import '../../theme/styles/reset';
@import '../../theme/styles/fonts';
@import '../../theme/styles/global';

/* SMenuAside */
.menu-enter {
  transform: translateX(-$aside-width);
}

.menu-enter-active {
  transition: all ease $animate--time;
}

.menu-enter-to {
  transform: translateX(0);
}

.menu-leave {
  transform: translateX(0);
}

.menu-leave-active {
  transition: all ease 2s;
}

.menu-leave-to {
  transform: translateX(-$aside-width);
}

.aside {
  @include z-index(menu);

  position: fixed;
  display: inline-flex;
  flex-direction: column;
  border: 0;
  height: 100%;
  max-height: 100vh;

  &__logo {
    border-right: 1px solid $color--border;
    border-bottom: 1px solid $color--border;
    padding: $size--20 $size--28;
    height: 88px;
    background-color: $color--white;
    transition: all ease $animate--time;

    &-icon {
      max-width: 190px;
    }
  }

  &__column {
    position: relative;
    z-index: 6;
    width: $aside-width;
    height: 100%;
    background-color: $color--white;
    transition: all ease $animate--time;

    &--children {
      position: absolute;
      top: 88px;

      &:nth-child(1) {
        z-index: 5;
      }

      &:nth-child(2) {
        z-index: 4;
      }

      &:nth-child(3) {
        z-index: 3;
      }

      &:nth-child(4) {
        z-index: 2;
      }

      &:nth-child(5) {
        z-index: 1;
      }
    }
  }
}

/* SMenuItem */
.menu-item {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: $size--12 $size--28;
  width: 100%;
  min-height: $size--48;
  text-decoration: none;
  color: $color--font;
  background-color: $color--white;
  transition: all ease $animate--time;
  cursor: pointer;

  &__text {
    display: flex;
    align-items: center;
  }

  &__arrow {
    color: $color--primary;
    transform: rotate(-90deg);
    transition: all ease $animate--time;
  }

  &:hover {
    color: $color--primary;
    background: $color--complimentary-bg-light;
  }

  &:hover:not(&--link) &__arrow {
    transform: rotate(-90deg) translateY(25%);
  }

  &--active {
    position: relative;
    font-weight: 800;
    color: $color--primary;
    background: $color--background;
  }

  &--link {
    margin-top: -1px;
    border-top: 1px solid $color--border;
    border-bottom: 1px solid $color--border;
    color: $color--primary;
  }
}

/* SMenuColumn */
.aside-menu-column {
  display: flex;
  flex-direction: column;
  overflow-x: hidden;
  overflow-y: auto;
  border-right: 1px solid $color--border;
  border-bottom: 1px solid $color--border;
  height: 100%;
  background-color: $color--white;

  &__header {
    display: flex;
    align-items: center;
    padding-right: $size--32;
    padding-left: $size--32;
    min-height: 68px;
    color: $color--font-muted;
    transition: all ease $animate--time;

    &-icon {
      transform: rotate(180deg);
      transition: all ease $animate--time;
    }

    &--active {
      color: $color--font;
      cursor: pointer;

      &:hover {
        color: $color--danger;
      }

      &:hover &-icon {
        transform: rotate(180deg) translateX(25%);
      }
    }
  }

  &__blocks {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: 100%;
  }
}
