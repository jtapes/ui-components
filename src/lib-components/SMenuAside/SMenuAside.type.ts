export type ElementColumnType = {
  name: string
  url?: string
  type?: string
  active?: boolean
  iconPrepend?: string
  iconAppend?: string
  data?: any
  child?: {
    menu: ElementColumnType[][]
    header?: ElementColumnType
  }
  parentColumn?: {
    menu: ElementColumnType[][]
    header?: ElementColumnType
  }
  parentElement?: ElementColumnType
  column?: {
    menu: ElementColumnType[][]
    header?: ElementColumnType
  }
}

export type ColumnType = {
  menu: ElementColumnType[][]
  header?: ElementColumnType & {
    close?: boolean
  }
}

export type ItemsMenuType = ColumnType & {
  staticBlockMenu?: ElementColumnType[]
}
