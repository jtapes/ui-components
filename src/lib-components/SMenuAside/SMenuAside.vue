<template>
  <aside class="aside us-n">
    <div
      v-if="logo"
      class="aside__logo"
      :style="`width: ${columnWidth * (columns.length + 1)}px;`"
    >
      <svg-icon class="aside__logo-icon" :name="logo" />
    </div>
    <transition-group name="menu" :css="isAnimation">
      <div
        v-for="i of columns.length"
        :key="i"
        class="aside__column aside__column--children"
        :style="`left: ${columnWidth * i}px;`"
      >
        <s-aside-column
          :items="columns[i - 1]"
          @element-click="elementClick"
          @element-hover="elementHover($event, i + 1)"
        />
      </div>
    </transition-group>
    <div v-if="headColumn" class="aside__column">
      <s-aside-column
        :items="headColumn"
        :header-disable="true"
        @element-click="elementClick"
        @element-hover="elementHover"
        @header-click="headerClick"
      />
    </div>
  </aside>
</template>

<script lang="ts">
import { Component, Vue, Prop, Watch } from 'vue-property-decorator'
import SAsideColumn from './SMenuColumn.vue'
import type { ColumnType, ElementColumnType, ItemsMenuType } from './SMenuAside.type'
import SvgIcon from '@/lib-components/SvgIcon/SvgIcon.vue'

@Component({
  components: {
    SAsideColumn,
    SvgIcon,
  },
})
export default class SMenuAside extends Vue {
  columns: ColumnType[] = []
  headColumn: ColumnType | null = null
  columnWidth = 320
  isAnimation = true

  /** Меню */
  @Prop({ type: Object, required: true })
  items!: ItemsMenuType

  /** Логотип */
  @Prop({ type: String })
  logo?: string

  created(): void {
    this.setParent(this.items)
    const elementActive: ElementColumnType | null = this.searchActive(
      this.items.menu
    )
    if (elementActive?.column) {
      this.setHeadColumn(elementActive.column)
    } else {
      this.setHeadColumn(this.items)
    }
  }

  @Watch('columns', { deep: true })
  watchColumn(): void {
    const isDefaultColumn = !this.headColumn?.header?.child
    const name = 'Меню'
    if (this.headColumn?.header && this.items?.header && isDefaultColumn) {
      if (this.columns.length > 0) {
        this.headColumn.header.name = name
        this.headColumn.header.close = true
      } else {
        this.headColumn.header.name = this.items.header.name
        this.headColumn.header.close = false
      }
    }
  }

  // записать в первую колонку
  setHeadColumn(column: ColumnType): void {
    // определяем является ли колонка самым верхним уровнем вложенности
    const isDefaultColumn = !column.header?.child
    let header = null
    if (column.header) {
      header = { ...column.header }
      if (this.items.header) {
        header.name = isDefaultColumn ? this.items.header.name : 'Меню'
      }
    }
    this.headColumn = {
      header: header || undefined,
      menu: this.items.staticBlockMenu
        ? [...column.menu, this.items.staticBlockMenu]
        : column.menu,
    }
  }

  // создание ссылок на родителей
  setParent(
    items: ItemsMenuType,
    parentColumn: null | ColumnType = null,
    parentElement: null | ElementColumnType = null
  ): void {
    if (items.menu) {
      for (const block of items.menu) {
        for (const element of block) {
          element.column = items
          if (parentColumn && parentElement) {
            element.parentColumn = parentColumn
            element.parentElement = parentElement
          }
          if (element.child) {
            element.child.header = element
            if (element.child.header) {
              this.setParent(element.child, items, element)
            }
          }
        }
      }
    }
  }

  // поиск последнего активного элемента
  searchActive(
    menu: ColumnType['menu'],
    elementActive: ElementColumnType | null = null
  ): null | ElementColumnType {
    for (const block in menu) {
      for (const keyEl in menu[block]) {
        if (menu[block][keyEl].active) {
          const child = menu[block][keyEl].child
          if (child) {
            return this.searchActive(child.menu, menu[block][keyEl])
          } else {
            return menu[block][keyEl]
          }
        }
      }
    }
    return elementActive
  }

  // выстраиваем меню активных элементов
  menuActive(menu: ColumnType['menu'], depth = 1): void{
    for (const block in menu) {
      for (const keyEl in menu[block]) {
        if (menu[block][keyEl].active && menu[block][keyEl].child) {
          const child = menu[block][keyEl].child
          if (child) {
            this.$set(this.columns, depth - 1, child)
            if (child.menu) {
              this.menuActive(child.menu, depth + 1)
            }
          }
        }
      }
    }
  }

  // отключить все активные элементы меню
  menuRemoveActive(menu: ColumnType['menu'] = this.items.menu): void {
    for (const block in menu) {
      for (const keyEl in menu[block]) {
        if (menu[block][keyEl].active) {
          menu[block][keyEl].active = false
          this.$set(menu[block], keyEl, menu[block][keyEl])
        }
        if (menu[block][keyEl].child) {
          this.menuRemoveActive(menu[block][keyEl].child?.menu)
        }
      }
    }
  }

  // построить путь из вложенности активных элементов
  menuAddActive(element: ElementColumnType, isHeadColumn = true): void {
    if (element.parentElement) {
      element.parentElement.active = true
      if (isHeadColumn && element.column) {
        this.setHeadColumn(element.column)
      }
      this.menuAddActive(element.parentElement, false)
    }
  }

  elementClick(element: ElementColumnType): void {
    this.isAnimation = false
    this.menuRemoveActive()
    element.active = true
    this.menuAddActive(element)
    this.columns = []
    this.$nextTick(() => {
      this.isAnimation = true
    })
    this.$emit('element-click', element)
  }

  elementHover(element: ElementColumnType, depth = 1): void {
    if (element.child && element.child.menu) {
      this.removeAnimateColumn(this.columns.length - depth)
      element.child.header = element
      this.$set(this.columns, depth - 1, element.child)
    }
  }

  headerClick(header: ElementColumnType): void {
    if (header.column && this.headColumn) {
      this.setHeadColumn(this.items)
      this.menuActive(this.headColumn.menu)
    } else {
      this.isAnimation = false
      this.columns = []
      this.$nextTick(() => {
        this.isAnimation = true
      })
    }
    this.$emit('header-click', header)
  }

  // закрытие колонок меню
  removeAnimateColumn(count = this.columns.length): void {
    if (count > 0) {
      this.columns.pop()
      setTimeout(() => {
        this.removeAnimateColumn(count - 1)
      }, 200)
    }
  }

  closeColumn(event: Event): void {
    if (event.target) {
      const target = event.target as HTMLElement
      if (!target.closest('.aside')) {
        this.removeAnimateColumn()
      }
    }
  }

  mounted(): void {
    document.addEventListener('click', this.closeColumn)
  }

  destroy(): void {
    document.removeEventListener('click', this.closeColumn)
  }
}
</script>
