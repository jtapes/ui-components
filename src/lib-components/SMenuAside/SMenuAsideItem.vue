<template>
  <div>
    <div
      v-if="type !== 'link'"
      class="menu-item"
      :class="{ 'menu-item--active': active }"
      v-on="$listeners"
    >
      <span class="menu-item__text">
        <svg-icon v-if="iconPrepend" :name="iconPrepend" class="icon-24 mr-8" />
        {{ name }}
      </span>
      <svg-icon
        v-if="iconAppend !== 'dropdown-24-32' || child"
        :name="iconAppend"
        class="icon-24"
        :class="{ 'menu-item__arrow': iconAppend === 'dropdown-24-32' }"
      />
    </div>
    <a v-else :href="url" class="menu-item menu-item--link">
      <span>{{ name }}</span>
      <svg-icon name="link-blank-24-32" class="icon-24 menu-item__arrow" />
    </a>
  </div>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator'
import SvgIcon from '@/lib-components/SvgIcon/SvgIcon.vue'

@Component({
  components: {
    SvgIcon
  }
})
export default class SAsideItem extends Vue {
  @Prop({ type: String, required: true })
  name!: string

  @Prop({ type: String })
  url?: string

  @Prop({ type: Boolean, default: false })
  active!: boolean

  @Prop({ type: String })
  iconPrepend?: string

  @Prop({ type: String, default: 'dropdown-24-32' })
  iconAppend!: string

  @Prop({ type: String, default: 'menu-item' })
  type!: string

  @Prop({ type: Object })
  child?: object
}
</script>
