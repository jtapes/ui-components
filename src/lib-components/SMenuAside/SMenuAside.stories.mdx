import { action } from '@storybook/addon-actions'
import { config, withDesign } from 'storybook-addon-designs'
import { withTests } from '@storybook/addon-jest';
import {
  Meta,
  Story,
  ArgsTable,
  Description,
  Source,
  Canvas,
} from '@storybook/addon-docs'
import results from '/.jest-test-results.json';
import SMenuAside from '@/lib-components/SMenuAside/SMenuAside.vue'

<Meta
  title="DESIGN SYSTEM/Atom/MenuAside"
  component={SMenuAside}
  decorators={[withDesign, withTests({results})]}
  argTypes={{
    items: {
      control: 'object',
      defaultValue: {
        header: {
          name: 'Предыдущая страница',
          url: '/',
        },
        menu: [
          [
            {
              name: 'Главная',
              iconAppend: 'burger',
            },
            {
              name: 'Заказы',
              active: true,
              child: {
                menu: [
                  [
                    {
                      name: 'О компании',
                    },
                    {
                      name: 'Клиентам',
                      active: true,
                      child: {
                        menu: [
                          [
                            {
                              name: 'Стратегия устойчивого развития',
                              child: {
                                menu: [
                                  [
                                    {
                                      name: 'Экологические проекты',
                                      child: {
                                        menu: [
                                          [
                                            {
                                              name: 'Изменение климата',
                                            },
                                            {
                                              name: 'Биоразнообразие',
                                            }
                                          ]
                                        ]
                                      }
                                    }
                                  ]
                                ]
                              }
                            }
                          ]
                        ]
                      }
                    },
                    {
                      name: 'Инвесторам',
                      child: {
                        menu: [
                          [
                            {
                              name: 'Устойчивое развитие',
                            }
                          ]
                        ]
                      }
                    }
                  ]
                ]
              },
            },
            {
              name: 'Финансы',
              child: {
                menu: [
                  [
                    {
                      name: 'Поддержка',
                    },
                  ]
                ]
              },
            },
            {
              name: 'Программа лояльности',
              iconPrepend: 'burger'
            },
            {
              name: 'Профиль',
            },
            {
              name: 'Компания',
              url: '/',
              type: 'link'
            },
          ]
        ],
        staticBlockMenu: [
          {
            name: 'Склад',
            url: '/',
            type: 'link'
          },
          {
            name: 'Аукционы',
            url: '/',
            type: 'link'
          },
          {
            name: 'Производство',
            url: '/',
            type: 'link'
          }
        ]
      }
    },
    logo: {
      control: 'string',
      defaultValue: 'logo'
    },
  }}
  parameters={{
    design: config({
      type: 'figma',
      url: 'https://www.figma.com/file/48lgW7NE3JB5dVC0Mec5o0/MVP-3-Account?node-id=41%3A1243',
    }),
    docs: {
      source: {
        type: 'code',
      },
    },
  }}
/>

export const Template = (args, {argTypes}) => ({
  props: Object.keys(argTypes),
  components: {SMenuAside},
  template: `<s-menu-aside style="position: fixed" :items='items' @element-click="element" @header-click="header" :logo="logo" />`,
  methods: {
    element(element) {
      action('element')(element)
    },
    header(element) {
      action('header')(element)
    },
  },
})

<Canvas withToolbar withSource="none">
  <Story name="SMenuAside">
    {Template.bind({})}
  </Story>
</Canvas>

<Description of={SMenuAside} />
<ArgsTable of={SMenuAside} />
<Source code={`<s-pagination :items="object" :logo="/" />`} />
