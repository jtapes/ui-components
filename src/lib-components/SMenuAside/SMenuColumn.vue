<template>
  <div class="aside-menu-column">
    <header
      v-if="items.header && items.header.name"
      class="aside-menu-column__header"
      :class="{ 'aside-menu-column__header--active': headerDisable }"
      @click="headerClick(items.header)"
    >
      <svg-icon
        v-if="headerDisable"
        :name="!items.header.close ? 'arrow-24-32' : 'close-24-32'"
        class="aside-menu-column__header-icon icon-24 mr-8 color-primary"
      />
      {{ items.header.name }}
    </header>
    <div class="aside-menu-column__blocks">
      <div
        v-for="(item, i) of items.menu"
        :key="i"
        class="aside-menu-column__block"
      >
        <s-menu-aside-item
          v-for="element in item"
          :key="element.name"
          v-bind="element"
          @click="elementClick(element)"
          @mouseover="elementHover(element)"
        />
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator'
import SMenuAsideItem from './SMenuAsideItem.vue'
import type { ColumnType, ElementColumnType } from './SMenuAside.type'
import SvgIcon from '@/lib-components/SvgIcon/SvgIcon.vue'

@Component({
  components: {
    SMenuAsideItem,
    SvgIcon,
  },
})
export default class SMenuAsideColumn extends Vue {
  @Prop({ type: Object, required: true })
  items!: ColumnType

  @Prop({ type: Boolean, default: false })
  headerDisable!: boolean

  elementClick(element: ElementColumnType): void {
    this.$emit('element-click', element)
  }

  elementHover(element: ElementColumnType): void {
    this.$emit('element-hover', element)
  }

  headerClick(element: ElementColumnType): void {
    this.$emit('header-click', element)
  }
}
</script>
