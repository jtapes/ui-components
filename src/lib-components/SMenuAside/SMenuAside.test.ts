import { mount } from '@vue/test-utils'
import SMenuAside from '@/lib-components/SMenuAside/SMenuAside.vue'

describe('MenuAside', () => {
  test('Init menu amd static', () => {
    const wrapper: any = mount(SMenuAside, {
      propsData: {
        logo: 'logo',
        items: {
          header: {
            name: 'Предыдущая страница',
            url: '/',
          },
          menu: [
            [
              {
                name: 'Главная',
                iconAppend: 'burger',
              },
              {
                name: 'Заказы',
                active: true,
              },
              {
                name: 'Финансы',
                child: {
                  menu: [
                    [
                      {
                        name: 'Поддержка',
                      },
                    ],
                  ],
                },
              },
              {
                name: 'Программа лояльности',
                iconPrepend: 'burger',
              },
              {
                name: 'Профиль',
              },
              {
                name: 'Компания',
                url: '/',
                type: 'link',
              },
            ],
          ],
          staticBlockMenu: [
            {
              name: 'Склад',
              url: '/',
              type: 'link',
            },
            {
              name: 'Аукционы',
              url: '/',
              type: 'link',
            },
            {
              name: 'Производство',
              url: '/',
              type: 'link',
            },
          ],
        },
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('Init menu active', () => {
    const wrapper: any = mount(SMenuAside, {
      propsData: {
        logo: 'logo',
        items: {
          header: {
            name: 'Предыдущая страница',
            url: '/',
          },
          menu: [
            [
              {
                name: 'Главная',
                iconAppend: 'burger',
              },
              {
                name: 'Заказы',
                active: true,
                child: {
                  menu: [
                    [
                      {
                        name: 'О компании',
                        active: true,
                      },
                    ],
                  ],
                },
              },
              {
                name: 'Финансы',
                child: {
                  menu: [
                    [
                      {
                        name: 'Поддержка',
                      },
                    ],
                  ],
                },
              },
              {
                name: 'Программа лояльности',
                iconPrepend: 'burger',
              },
              {
                name: 'Профиль',
              },
              {
                name: 'Компания',
                url: '/',
                type: 'link',
              },
            ],
          ],
          staticBlockMenu: [
            {
              name: 'Склад',
              url: '/',
              type: 'link',
            },
            {
              name: 'Аукционы',
              url: '/',
              type: 'link',
            },
            {
              name: 'Производство',
              url: '/',
              type: 'link',
            },
          ],
        },
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('Menu hover click event', async () => {
    const wrapper: any = mount(SMenuAside, {
      propsData: {
        logo: 'logo',
        items: {
          header: {
            name: 'Предыдущая страница',
            url: '/',
          },
          menu: [
            [
              {
                name: 'Главная',
                iconAppend: 'burger',
              },
              {
                name: 'Заказы',
                active: true,
                child: {
                  menu: [
                    [
                      {
                        name: 'О компании',
                        active: true,
                      },
                    ],
                  ],
                },
              },
              {
                name: 'Финансы',
                child: {
                  menu: [
                    [
                      {
                        name: 'Поддержка',
                      },
                    ],
                  ],
                },
              },
              {
                name: 'Программа лояльности',
                iconPrepend: 'burger',
              },
              {
                name: 'Профиль',
              },
              {
                name: 'Компания',
                url: '/',
                type: 'link',
              },
            ],
          ],
        },
      },
    })

    const hoverElement = wrapper.find('.aside-menu-column__header')
    expect(hoverElement.text()).toBe('Меню')
    await hoverElement.trigger('click')
    expect(hoverElement.text()).toBe('Меню')
    expect(wrapper.findAll('.aside__column--children').length).toBe(1)

    await hoverElement.trigger('click')
    expect(hoverElement.text()).toBe('Предыдущая страница')
    expect(wrapper.findAll('.aside__column--children').length).toBe(0)

    const element = wrapper.findAll('.menu-item').at(2) // финансы
    await element.trigger('mouseover')
    expect(wrapper.findAll('.aside__column--children').length).toBe(1)
  })
})
