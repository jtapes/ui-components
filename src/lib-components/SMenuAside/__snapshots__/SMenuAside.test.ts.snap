// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuAside Init menu active 1`] = `
<aside
  class="aside us-n"
>
  <div
    class="aside__logo"
    style="width: 320px;"
  >
    <svg
      class="svg-icon aside__logo-icon svg-icon--logo"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        href="#logo"
        xlink:href="#logo"
      />
    </svg>
  </div>
   
  <transition-group-stub
    css="true"
    name="menu"
  />
   
  <div
    class="aside__column"
  >
    <div
      class="aside-menu-column"
    >
      <header
        class="aside-menu-column__header aside-menu-column__header--active"
      >
        <svg
          class="svg-icon aside-menu-column__header-icon icon-24 mr-8 color-primary svg-icon--arrow-24-32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#arrow-24-32"
            xlink:href="#arrow-24-32"
          />
        </svg>
        
    Меню
  
      </header>
       
      <div
        class="aside-menu-column__blocks"
      >
        <div
          class="aside-menu-column__block"
        >
          <div
            column="[object Object]"
            parentcolumn="[object Object]"
            parentelement="[object Object]"
          >
            <div
              class="menu-item menu-item--active"
            >
              <span
                class="menu-item__text"
              >
                <!---->
                
      О компании
    
              </span>
               
              <!---->
            </div>
          </div>
        </div>
        <div
          class="aside-menu-column__block"
        >
          <div>
            <a
              class="menu-item menu-item--link"
              href="/"
            >
              <span>
                Склад
              </span>
               
              <svg
                class="svg-icon icon-24 menu-item__arrow svg-icon--link-blank-24-32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  href="#link-blank-24-32"
                  xlink:href="#link-blank-24-32"
                />
              </svg>
            </a>
          </div>
          <div>
            <a
              class="menu-item menu-item--link"
              href="/"
            >
              <span>
                Аукционы
              </span>
               
              <svg
                class="svg-icon icon-24 menu-item__arrow svg-icon--link-blank-24-32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  href="#link-blank-24-32"
                  xlink:href="#link-blank-24-32"
                />
              </svg>
            </a>
          </div>
          <div>
            <a
              class="menu-item menu-item--link"
              href="/"
            >
              <span>
                Производство
              </span>
               
              <svg
                class="svg-icon icon-24 menu-item__arrow svg-icon--link-blank-24-32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  href="#link-blank-24-32"
                  xlink:href="#link-blank-24-32"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</aside>
`;

exports[`MenuAside Init menu amd static 1`] = `
<aside
  class="aside us-n"
>
  <div
    class="aside__logo"
    style="width: 320px;"
  >
    <svg
      class="svg-icon aside__logo-icon svg-icon--logo"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        href="#logo"
        xlink:href="#logo"
      />
    </svg>
  </div>
   
  <transition-group-stub
    css="true"
    name="menu"
  />
   
  <div
    class="aside__column"
  >
    <div
      class="aside-menu-column"
    >
      <header
        class="aside-menu-column__header aside-menu-column__header--active"
      >
        <svg
          class="svg-icon aside-menu-column__header-icon icon-24 mr-8 color-primary svg-icon--arrow-24-32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#arrow-24-32"
            xlink:href="#arrow-24-32"
          />
        </svg>
        
    Предыдущая страница
  
      </header>
       
      <div
        class="aside-menu-column__blocks"
      >
        <div
          class="aside-menu-column__block"
        >
          <div
            column="[object Object]"
          >
            <div
              class="menu-item"
            >
              <span
                class="menu-item__text"
              >
                <!---->
                
      Главная
    
              </span>
               
              <svg
                class="svg-icon icon-24 svg-icon--burger"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  href="#burger"
                  xlink:href="#burger"
                />
              </svg>
            </div>
          </div>
          <div
            column="[object Object]"
          >
            <div
              class="menu-item menu-item--active"
            >
              <span
                class="menu-item__text"
              >
                <!---->
                
      Заказы
    
              </span>
               
              <!---->
            </div>
          </div>
          <div
            column="[object Object]"
          >
            <div
              class="menu-item"
            >
              <span
                class="menu-item__text"
              >
                <!---->
                
      Финансы
    
              </span>
               
              <svg
                class="svg-icon icon-24 svg-icon--dropdown-24-32 menu-item__arrow"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  href="#dropdown-24-32"
                  xlink:href="#dropdown-24-32"
                />
              </svg>
            </div>
          </div>
          <div
            column="[object Object]"
          >
            <div
              class="menu-item"
            >
              <span
                class="menu-item__text"
              >
                <svg
                  class="svg-icon icon-24 mr-8 svg-icon--burger"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <use
                    href="#burger"
                    xlink:href="#burger"
                  />
                </svg>
                
      Программа лояльности
    
              </span>
               
              <!---->
            </div>
          </div>
          <div
            column="[object Object]"
          >
            <div
              class="menu-item"
            >
              <span
                class="menu-item__text"
              >
                <!---->
                
      Профиль
    
              </span>
               
              <!---->
            </div>
          </div>
          <div
            column="[object Object]"
          >
            <a
              class="menu-item menu-item--link"
              href="/"
            >
              <span>
                Компания
              </span>
               
              <svg
                class="svg-icon icon-24 menu-item__arrow svg-icon--link-blank-24-32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  href="#link-blank-24-32"
                  xlink:href="#link-blank-24-32"
                />
              </svg>
            </a>
          </div>
        </div>
        <div
          class="aside-menu-column__block"
        >
          <div>
            <a
              class="menu-item menu-item--link"
              href="/"
            >
              <span>
                Склад
              </span>
               
              <svg
                class="svg-icon icon-24 menu-item__arrow svg-icon--link-blank-24-32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  href="#link-blank-24-32"
                  xlink:href="#link-blank-24-32"
                />
              </svg>
            </a>
          </div>
          <div>
            <a
              class="menu-item menu-item--link"
              href="/"
            >
              <span>
                Аукционы
              </span>
               
              <svg
                class="svg-icon icon-24 menu-item__arrow svg-icon--link-blank-24-32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  href="#link-blank-24-32"
                  xlink:href="#link-blank-24-32"
                />
              </svg>
            </a>
          </div>
          <div>
            <a
              class="menu-item menu-item--link"
              href="/"
            >
              <span>
                Производство
              </span>
               
              <svg
                class="svg-icon icon-24 menu-item__arrow svg-icon--link-blank-24-32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  href="#link-blank-24-32"
                  xlink:href="#link-blank-24-32"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</aside>
`;
